source/common/stats/thread_local_store.cc

```diff
@@ -182,7 +183,8 @@ StatType& ThreadLocalStoreImpl::ScopeImpl::safeMakeStat(
     if (stat == nullptr) {
       parent_.num_last_resort_stats_.inc();
       stat =
-          make_stat(parent_.heap_allocator_, name, std::move(tag_extracted_name), std::move(tags));
+          make_stat(parent_.heap_allocator_, name.substr(0, parent_.statsOptions().maxNameLength()),
```

Comment:
This seems fishy.  Since we now throw if the heap allocator fails to allocate, this statement will never be called in the non-hot restart case.  However, it's still an embedded assumption about the specific behavior of all allocators - fixed length if they can fail and non-fixed length if they cannot fail.  Is there any problem with changing the hot restart allocator to perform the fallback rather than relying on the caller?  Since this is technically correct, I think it's fine to leave it this way for now, but please toss in a TODO to get rid of this assumption.

Reply:
Added this todo in cb0fd8a.